### PR TITLE
[MINOR][DOC] Fix some typos and grammar issues

### PR DESCRIPTION
**Pull request by dsakuma**
## What changes were proposed in this pull request?

Easy fix in the documentation.

## How was this patch tested?

N/A

Closes #20948

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
that's fine but mind taking a look for other typoes while we are here?

**Comment by dsakuma** [comment]
@HyukjinKwon Great idea! I've found and fixed some other issues using a spell checker.

**Comment by jerryshao** [comment]
It would be better to check other docs, not only in configurations here.

**Review by felixcheung — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
yea, it's fine as is to me but it's definitely better to check other docs while we are here.

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by dsakuma** [comment]
I'm gonna check other docs.

**Comment by HyukjinKwon** [comment]
Hey @dsakuma, have you had a change to look other docs too? if you are busy, let me just go and merge this. BTW, mind adding `Closes #20948` in the PR description too so that that PR can be automatically closed when this PR is merged? That PR includes the same change included here.

**Comment by dsakuma** [comment]
@HyukjinKwon I think I'm gonna fininsh today. And I've added the issue in the description.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
test this please

**Comment by dsakuma** [comment]
@HyukjinKwon I think that's it! Please let me know if there's anything else I can do.

**Comment by HyukjinKwon** [comment]
Seems there's some problems with Jenkins. Will make the tests run and merge it later.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
@dsakuma, mind if I ask to fix the PR title to .. `[MINOR][DOC] ...` just to consistent with other PRs? It's not a small typo anymore :). Thanks for your effort.

**Comment by dsakuma** [comment]
@HyukjinKwon I've fixed the title format :D

File: docs/ml-collaborative-filtering.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ above) and "drop". Further strategies may be supported in future.
 <div class="codetabs">
 <div data-lang="scala" markdown="1">
 
-In the following example, we load ratings data from the
+In the following example, we load rating data from the
```
Comment (HyukjinKwon) [inline docs/ml-collaborative-filtering.md]: @dsakuma, ratings seems fine (also given the link http://grouplens.org/datasets/movielens/)

File: docs/mllib-feature-extraction.md [hunk 2]
```diff
@@ -105,7 +105,7 @@ p(w_i | w_j ) = \frac{\exp(u_{w_i}^{\top}v_{w_j})}{\sum_{l=1}^{V} \exp(u_l^{\top
 \]`
 where $V$ is the vocabulary size. 
 
-The skip-gram model with softmax is expensive because the cost of computing $\log p(w_i | w_j)$ 
+The skip-gram model with softmax is expensive because of the cost of computing $\log p(w_i | w_j)$
```
Comment (HyukjinKwon) [inline docs/mllib-feature-extraction.md]: seems a mistake.

File: docs/mllib-naive-bayes.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ These models are typically used for [document classification](http://nlp.stanfor
 Within that context, each observation is a document and each
 feature represents a term whose value is the frequency of the term (in multinomial naive Bayes) or
 a zero or one indicating whether the term was found in the document (in Bernoulli naive Bayes).
-Feature values must be nonnegative. The model type is selected with an optional parameter
+Feature values must be non-negative. The model type is selected with an optional parameter
```
Comment (HyukjinKwon) [inline docs/mllib-naive-bayes.md]: seems the previous one also correct.

File: sql/README.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ This module provides support for executing relational queries expressed in eithe
 Spark SQL is broken up into four subprojects:
  - Catalyst (sql/catalyst) - An implementation-agnostic framework for manipulating trees of relational operators and expressions.
  - Execution (sql/core) - A query planner / execution engine for translating Catalyst's logical query plans into Spark RDDs.  This component also includes a new public interface, SQLContext, that allows users to execute SQL or LINQ statements against existing RDDs and Parquet files.
- - Hive Support (sql/hive) - Includes an extension of SQLContext called HiveContext that allows users to write queries using a subset of HiveQL and access data from a Hive Metastore using Hive SerDes.  There are also wrappers that allows users to run queries that include Hive UDFs, UDAFs, and UDTFs.
+ - Hive Support (sql/hive) - Includes an extension of SQLContext called HiveContext that allows users to write queries using a subset of HiveQL and access data from a Hive Metastore using Hive SerDes. There are also wrappers that allow  users to run queries that include Hive UDFs, UDAFs, and UDTFs.
```
Comment (HyukjinKwon) [inline sql/README.md]: There seems an extra place after `allow`.

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**